source/SIL.AppBuilder.Portal.Frontend/src/ui/components/inputs/user-select/with-data.tsx

```diff
@@ -1,117 +1,169 @@
+  currentUsersGroupMemberships: GroupMembershipResource[];
+  usersFromCache: UserResource[];
+  currentUser: UserResource;
   selected: Id;
   groupId: Id;
   restrictToGroup: boolean;
+  scopeToOrganization?: OrganizationResource;
 }
 
 type IProps =
   & IOwnProps
   & WithDataProps;
 
-export function withData(WrappedComponent) {
-  const mapNetworkToProps = (passedProps) => {
-    return {
-      users: [
-        q => q.findRecords(USER), {
-          label: 'Get Users for User Input Select',
-          sources: {
-            remote: {
-              settings: { ...defaultSourceOptions() },
-              include: ['group-memberships'],
-            }
-          }
-        }
-      ]
-    };
-  };
-
-  const mapRecordsToProps = (passedProps) => {
-    const { type, id } = passedProps.currentUser;
+interface IState {
+  users?: UserResource[];
+}
 
-    return {
-      currentUsersGroupMemberships: q => q.findRelatedRecords({ type, id }, 'groupMemberships'),
-      groupMemberships: q => q.findRecords('groupMembership')
-    };
-  };
+export function withData(WrappedComponent) {
+  class DataWrapper extends React.Component<IProps, IState> {
+    isFilteringUsers = false;
+    state: IState = {};
 
-  class DataWrapper extends React.Component<IProps> {
-    render() {
+    determineAvailableUsers = async () => {
       const {
+        dataStore,
         users,
-        currentUser,
         selected,
         groupId,
-        groupMemberships,
+        scopeToOrganization,
         restrictToGroup,
-        currentUsersGroupMemberships,
-        ...otherProps
       } = this.props;
 
-      if (!users || !groupMemberships || !currentUsersGroupMemberships) {
-        return <Loader />;
+      // remove users who _can't_ be assigned to this project
+      // due to not having organization overlap
+      const filtered = [];
+      const promises = (users || []).map(async user => {
+        let isInOrganization = true;
+        let isInGroup = true;
+
+        if (scopeToOrganization) {
+          const organizationId = scopeToOrganization.id;
+          const organizations = await retrieveRelation(dataStore, [user, 'organizationMemberships', 'organization']);
+          const ids = (organizations || []).map(o => o && o.id);
+
+          isInOrganization = ids.includes(organizationId);
+        }
+
+        if (restrictToGroup && groupId) {
+          const groups = await retrieveRelation(dataStore, [user, 'groupMemberships', 'group']);
+          const ids = (groups || []).map(g => g && g.id);
+
+          isInGroup = ids.includes(groupId);
+        }
+
+        if (isInOrganization && isInGroup) {
+          filtered.push(user);
+        }
+      });
+
+      await Promise.all(promises);
+
+      // ensure the project owner is in the list
+      const isOwnerInList = filtered.find(user => user.id === selected);
+
+      if( (!isOwnerInList)) {
+        const owner = users.find(user => user.id === selected);
```

Comment:
find can return undefined when not found. suggest to add some sort of validation here

Reply:
I think for now, I'd rather let this error so we can figure ouwn why the selected user isn't in the users array.
I think it _shouldn't_ happen